Northstar.Client/mod/scripts/vscripts/ui/menu_mod_settings.nut

```diff
@@ -0,0 +1,1101 @@
+
+	file.modPanels = GetElementsByClassname( file.menu, "ModButton" )
+
+	AddMenuEventHandler( file.menu, eUIEvent.MENU_OPEN, OnModMenuOpened )
+	AddMenuEventHandler( file.menu, eUIEvent.MENU_CLOSE, OnModMenuClosed )
+
+	int len = file.modPanels.len()
+	for (int i = 0; i < len; i++)
+	{
+		
+		//AddButtonEventHandler( button, UIE_CHANGE, OnSettingButtonPressed  )
+		// get panel
+		var panel = file.modPanels[i]
+
+		// reset to default nav
+		var child = Hud_GetChild( panel, "BtnMod" )
+
+
+		child.SetNavUp( Hud_GetChild( file.modPanels[ GetIndex( i - 1, len ) ], "BtnMod" ) )
+		child.SetNavDown( Hud_GetChild( file.modPanels[ GetIndex( i + 1, len ) ], "BtnMod" ) )
+
+		// Enum button nav
+		child = Hud_GetChild( panel, "EnumSelectButton" )
+		Hud_DialogList_AddListItem( child, SETTING_ITEM_TEXT, "main" )
+		Hud_DialogList_AddListItem( child, SETTING_ITEM_TEXT, "next" )
+		Hud_DialogList_AddListItem( child, SETTING_ITEM_TEXT, "prev" )
+
+		child.SetNavUp( Hud_GetChild( file.modPanels[ GetIndex( i - 1, len ) ], "EnumSelectButton" ) )
+		child.SetNavDown( Hud_GetChild( file.modPanels[ GetIndex( i + 1, len ) ], "EnumSelectButton" ) )
+		Hud_AddEventHandler( child, UIE_CLICK, UpdateEnumSetting )
+
+		// reset button nav
+		
+		child = Hud_GetChild( panel, "ResetModToDefault" )
+
+		child.SetNavUp( Hud_GetChild( file.modPanels[ GetIndex( i - 1, len ) ], "ResetModToDefault" ) )
+		child.SetNavDown( Hud_GetChild( file.modPanels[ GetIndex( i + 1, len ) ], "ResetModToDefault" ) )
+
+		Hud_AddEventHandler( child, UIE_CLICK, ResetConVar )
+		
+		// text field nav
+		child = Hud_GetChild( panel, "TextEntrySetting" )
+
+		// 
+		Hud_AddEventHandler( child, UIE_LOSE_FOCUS, SendTextPanelChanges )
+
+		child.SetNavUp( Hud_GetChild( file.modPanels[ GetIndex( i - 1, len ) ], "TextEntrySetting" ) )
+		child.SetNavDown( Hud_GetChild( file.modPanels[ GetIndex( i + 1, len ) ], "TextEntrySetting" ) )
+
+		child = Hud_GetChild( panel, "Slider" )
+
+		child.SetNavUp( Hud_GetChild( file.modPanels[ GetIndex( i - 1, len ) ], "Slider" ) )
+		child.SetNavDown( Hud_GetChild( file.modPanels[ GetIndex( i + 1, len ) ], "Slider" ) )
+
+		file.sliders.append(MS_Slider_Setup(child))
+
+		Hud_AddEventHandler( child, UIE_CHANGE, OnSliderChange )
+
+		child = Hud_GetChild( panel, "OpenCustomMenu" )
+
+		Hud_AddEventHandler( child, UIE_CLICK, CustomButtonPressed )
+	}
+
+	//Hud_AddEventHandler( Hud_GetChild( file.menu, "BtnModsSearch" ), UIE_LOSE_FOCUS, OnFilterTextPanelChanged )
+	Hud_AddEventHandler( Hud_GetChild( file.menu, "BtnFiltersClear" ), UIE_CLICK, OnClearButtonPressed )
+	// mouse delta 
+	AddMouseMovementCaptureHandler( file.menu, UpdateMouseDeltaBuffer )
+
+	thread SearchBarUpdate()
+}
+
+void function SearchBarUpdate()
+{
+	while (true)
+	{
+		if (file.filterText != Hud_GetUTF8Text( Hud_GetChild( file.menu, "BtnModsSearch" ) ) )
+		{
+			file.filterText = Hud_GetUTF8Text( Hud_GetChild( file.menu, "BtnModsSearch" ) )
+			OnFiltersChange(0)
+		}
+		WaitFrame()
```

Comment:
You can use `AddButtonEvenHandler` like this: [menu_ns_serverbrowser.nut#207](https://github.com/R2Northstar/NorthstarMods/blob/main/Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut#L207) instead of a threaded while loop